src/censure.jl

```diff
@@ -0,0 +1,236 @@
+abstract BiFilter
+
+type BoxFilter <: BiFilter
+
+	scale :: Integer
+	in_length :: Integer
+	out_length :: Integer
+	in_area :: Float64
+	out_area :: Float64
+	in_weight :: Float64
+	out_weight :: Float64
+
+end
+
+type OctagonFilter <: BiFilter
+	
+	m_out :: Integer
+	m_in :: Integer
+	n_out :: Integer
+	n_in :: Integer
+	in_area :: Float64
+	out_area :: Float64
+	in_weight :: Float64
+	out_weight :: Float64
+
+end
+
+type CENSURE <: Detector
+
+	smallest :: Integer
+	largest :: Integer
+	filter_type :: Type
+	filter_stack :: Array
```

Comment:
`Array`, `Type`, and `Number` are also abstract. Use parametric types where necessary, e.g.:
```jl
type CENSURE{F,T,N} <: Detector
    smallest::Int
    largest::Int
    filter_type::F
    filter_stack::Array{T,N}
    responseThreshold::T
    lineThreshold::T
end
```
(There's a curious mix of underscores and camelcase here.)